src/data/finset/basic.lean

```diff
@@ -1895,6 +1895,11 @@ by { subst h, simp }
 theorem map_map {g : β ↪ γ} : (s.map f).map g = s.map (f.trans g) :=
 eq_of_veq $ by simp only [map_val, multiset.map_map]; refl
 
+lemma map_comm {β'} {f : β ↪ γ} {g : α ↪ β} {f' : α ↪ β'} {g' : β' ↪ γ}
+  (h_comm : ∀ a, f (g a) = g' (f' a)) :
+  (s.map g).map f = (s.map f').map g' :=
```

Comment:
I think this probably makes more sense as
```suggestion
  (s.map g).map f = (s.map g).map f :=
```
Or even
```suggestion
  function.commute (map g) (map f) :=
```

Reply:
This is strictly less general, though. And it does not match [`set.image_comm`](https://leanprover-community.github.io/mathlib_docs/data/set/basic.html#set.image_comm).

Reply:
I'll make a PR to fix image_comm then :)

I don't think losing generality matters; we already have `image_image` as the most general form.

Reply:
... and `filter.map_comm`, and all the algebraic replacement rules in `data.set.basic`, `data.finset.n_ary`, `data.filter.n_ary`... It actually does matter. The greater generality is required when proving associativity/commutativity/distributivity of pointwise operations.

Reply:
I think those cases only need the generality of `function.semiconj`?

Reply:
I guess my objection is reduces to the fact that `_comm` usually refers to basic commutativity, rather than "this square is commutative"

Reply:
Sure, but in this context we already widely use it for "this square is commutative", so I don't see a reason to change.

Reply:
Yep, you've convinced me by pointing at `filter.map_comm` which was written by someone else